pkg/data/helpers.go

```diff
@@ -37,3 +45,52 @@ func getFiltersFromStruct(i interface{}, keyName *string) (*expression.KeyCondit
 	}
 	return kb, cb
 }
+
+func putItem(input interface{}, data interface{}, tableName string, expr *expression.Expression) error {
+
+	returnValue := "NONE"
+	i, _ := input.(*Account)
+	account, _ := data.(model.Account)
+
+	putMap, _ := dynamodbattribute.Marshal(account)
+	err := Invoke(i.DynamoDB, "PutItem",
+		&dynamodb.PutItemInput{
+			// Query in input Table
+			TableName: aws.String(tableName),
+			// Find record for the requested input
+			Item: putMap.M,
+			// Condition Expression
+			ConditionExpression:       expr.Condition(),
+			ExpressionAttributeNames:  expr.Names(),
+			ExpressionAttributeValues: expr.Values(),
+			// Return the updated record
+			ReturnValues: aws.String(returnValue),
+		},
+	)
+	var awsErr awserr.Error
+	if gErrors.As(err, &awsErr) {
+		if awsErr.Code() == "ConditionalCheckFailedException" {
+			return errors.NewConflict(
+				tableName,
+				"input",
+				fmt.Errorf("unable to update %s: Table has been modified since request was made", tableName))
+		}
+	}
+
+	return nil
+}
+
+func Invoke(any interface{}, name string, args... interface{}) error {
```

Comment:
I'm having trouble understanding the purpose of this function. Why not just use normal syntax for calling the PutItem method?
```
	_, err = a.DynamoDB.PutItem(	
		&dynamodb.PutItemInput{	
			// Query in Lease Table	
			TableName: aws.String(a.TableName),	
			// Find Account for the requested accountId	
			Item: putMap.M,	
			// Condition Expression	
			ConditionExpression:       expr.Condition(),	
			ExpressionAttributeNames:  expr.Names(),	
			ExpressionAttributeValues: expr.Values(),	
			// Return the updated record	
			ReturnValues: aws.String(returnValue),	
		},	
	)
```

It looks like the only thing gained here is handling the error inside of this helper function rather than in the calling function

Reply:
Hi Josh, I was trying to do error handling inside the putItem. I have made it simpler now.